net-irc/irssi/patches/irssi-0.8.17-haiku.patch

```diff
@@ -0,0 +1,315 @@
+From e9617d64b8b8846dd7c7379cab59fe6ee85ca629 Mon Sep 17 00:00:00 2001
+From: pisculichi <pisculichi@openmailbox.org>
+Date: Fri, 18 Dec 2015 00:12:26 +0000
+Subject: [PATCH] Add support for ncurses in Haiku
+
+---
+ m4/curses.m4 | 20 +++++++++++++++++++-
+ 1 file changed, 19 insertions(+), 1 deletion(-)
+
+diff --git a/m4/curses.m4 b/m4/curses.m4
+index 82b110f..1def9e9 100644
+--- a/m4/curses.m4
++++ b/m4/curses.m4
+@@ -131,7 +131,7 @@ AC_DEFUN([AC_CHECK_CURSES],[
+ 	  if test x$withval = xno ; then
+ 		search_ncurses=false
+ 	  elif test x$withval != xyes ; then
+-		AC_NCURSES($withval/include, ncurses.h, -L$withval/lib -lncurses, -I$withval/include, "ncurses on $withval/include")
++		AC_NCURSES($withval/include, ncurses.h, -L$withval/lib -lncurses, -I$withval/include, [ncurses on $withval/include])
+ 	  fi
+ 	)
+
+@@ -244,6 +244,24 @@ AC_DEFUN([AC_SEARCH_NCURSES], [
+ 	       [renamed ncurses in /usr/local/include/ncurses])
+     AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses, -I/usr/include/ncurses -DRENAMED_NCURSES,
+ 	       [renamed ncurses in /usr/include/ncurses])
++
++    dnl
++    dnl Try Haiku ncurses
++    dnl Depending if we are building for the primary or secondary arch
++    dnl it can be installed at different locations
++    dnl
++    if $search_ncurses
++    then
++	if test -d /system/develop/headers
++	then
++	    haiku_arch=`getarch`
++	    for haiku_h in `findpaths -a $haiku_arch B_FIND_PATH_HEADERS_DIRECTORY`
++	    do
++		AC_NCURSES($haiku_h, ncurses.h, -lncurses,,
++		    [ncurses in $haiku_h])
++	    done
++        fi
++    fi
+
+     dnl
+     dnl We couldn't find ncurses, try SysV curses
+--
+2.2.2
+
+From e610ab583827cef07a92b354d07a5bb3f1a2e05d Mon Sep 17 00:00:00 2001
+From: pisculichi <pisculichi@openmailbox.org>
+Date: Fri, 18 Dec 2015 00:29:23 +0000
+Subject: [PATCH] Fix variable declarations to support ANSI C
+
+---
+ src/fe-common/core/fe-messages.c |  6 ++++--
+ src/fe-common/core/themes.c      | 14 +++++++++-----
+ src/fe-text/statusbar.c          | 27 ++++++++++++++++-----------
+ 3 files changed, 29 insertions(+), 18 deletions(-)
+
+diff --git a/src/fe-common/core/fe-messages.c b/src/fe-common/core/fe-messages.c
+index 97f3d84..2121653 100644
+--- a/src/fe-common/core/fe-messages.c
++++ b/src/fe-common/core/fe-messages.c
+@@ -369,6 +369,8 @@ static void sig_message_quit(SERVER_REC *server, const char *nick,
+ 	GString *chans;
+ 	GSList *tmp, *windows;
+ 	char *print_channel;
++	CHANNEL_REC *rec;
++	QUERY_REC *query;
+ 	int once, count, level = MSGLEVEL_QUITS;
```

Comment:
yes I know, but to support gcc2 untill gets removed from haiku. Sorry for my inglish